bmp-string.go

```diff
@@ -12,8 +13,12 @@ func bmpString(s string) ([]byte, error) {
 	//	  EncodeRune returns 0xfffd if the rune does not need special encoding
 	//  - the above RFC provides the info that BMPStrings are NULL terminated.
 
-	rv := make([]byte, 0, 2*len(s)+2)
-	for _, c := range []rune(s) {
+	rv := make([]byte, 0, 2*len(utf8String)+2)
```

Comment:
I'm ok with wasting a little here. The objective was to allocate at least enough to decourage any reallocation and potential copy of partial passwords. I know we don't have control over what the runtime does, but I can still try to give it hints... :smile: